src/libraries/System.Drawing.Common/src/System/Drawing/Printing/PrinterSettings.cs

```diff
@@ -206,7 +206,7 @@ public static StringCollection InstalledPrinters
                 {
                     // The printer name is at offset 0
                     //
-                    IntPtr namePointer = (IntPtr)Marshal.ReadIntPtr((IntPtr)(checked((long)buffer + i * sizeofstruct)));
+                    IntPtr namePointer = (IntPtr)Marshal.ReadIntPtr(checked((IntPtr)((long)buffer + i * sizeofstruct)));
```

Comment:
```suggestion
                    IntPtr namePointer = *(IntPtr*)(buffer + i * sizeofstruct);
```
Nothing in this method should be using `checked`. Convert it to use pointers. Also change type of `buffer` to `byte*`.